vmdb/app/models/sysprep_file.rb

```diff
@@ -0,0 +1,26 @@
+class SysprepFile
+  attr_reader :content
+
+  def initialize(content)
+    @content = content.respond_to?(:read) ? content.read : content
+    validate_content
+  end
+
+  private
+
+  def validate_content
+    send("validate_sysprep_#{content.include?("<?xml") ? "xml" : "ini"}")
+  end
+
+  def validate_sysprep_xml
+    name = Nokogiri::XML(content).root.try(:name)
+    raise "Invalid file contents detected" unless name == "unattend"
+  end
+
+  def validate_sysprep_ini
+    require 'inifile'
+    IniFile.new(:content => content)
+  rescue IniFile::Error
+    raise "Invalid file contents detected"
```

Comment:
@brandondunne 

Do we really want to throw away the more specific `IniFile::Error` exception and raise a generic StandardError?

If we want a more useful message, we should do something like this:
http://stackoverflow.com/a/4789702

```ruby
def validate_sysprep_ini
  require 'inifile'
  IniFile.new(:content => content)
rescue IniFile::Error => e
  raise e, "Invalid file contents detected. #{e}", e.backtrace
end
```